yang/frr-bgp-common-structure.yang

```diff
@@ -0,0 +1,807 @@
+                   is exceeded.";
+              }
+            }
+
+            case restart {
+              leaf restart-timer {
+                type uint16;
+                units "minutes";
+                description
+                  "Time interval in seconds after which the BGP session is
+                   re-established after being torn down due to exceeding the
+                   max-prefix limit.";
+              }
+            }
+
+            case threshold {
+              leaf shutdown-threshold-pct {
+                type bt:percentage;
+                description
+                  "Threshold on number of prefixes that can be received from
+                   a neighbour before generation of warning messages or log
+                   entries. Expressed as a percentage of max-prefixes.";
+              }
+            }
+
+            case threshold-restart {
+              leaf tr-shutdown-threshold-pct {
+                type bt:percentage;
+                description
+                  "Threshold on number of prefixes that can be received from
+                   a neighbour before generation of warning messages or log
+                   entries. Expressed as a percentage of max-prefixes.";
+              }
+
+              leaf tr-restart-timer {
+                type uint32;
+                units "minutes";
+                description
+                  "Time interval in seconds after which the BGP session is
+                   re-established after being torn down due to exceeding the
+                   max-prefix limit.";
+              }
+            }
+
+            case threshold-warning {
+              leaf tw-shutdown-threshold-pct {
+                type bt:percentage;
+                description
+                  "Threshold on number of prefixes that can be received from
+                   a neighbour before generation of warning messages or log
+                   entries. Expressed as a percentage of max-prefixes.";
+              }
+
+              leaf tw-warning-only {
+                type boolean;
+                default "false";
+                description
+                  "When set to 'true' only give warning message when limit
+                   is exceeded.";
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  grouping structure-neighbor-nexthop-self {
+    container nexthop-self {
+      description
+        "Parameters relating to the nexthop-self for the AFI-SAFI.";
+      leaf next-hop-self {
+        type boolean;
+        default "false";
+        description
+          "When set to 'true', EBGP learned routes are announced with the
+           local speaker's nexthop.";
+      }
+
+      leaf next-hop-self-force {
+        type boolean;
+        default "false";
+        description
+          "When set to 'true', EBGP learned routes are announced with the
+           local speaker's nexthop.";
+      }
```

Comment:
Do we need a separate setting for `next-hop-self-all`?

Reply:
At the backend, We have two FLAGS
#define PEER_FLAG_NEXTHOP_SELF              (1 << 2) /* next-hop-self */
#define PEER_FLAG_FORCE_NEXTHOP_SELF        (1 << 17) /* next-hop-self force */

For both leaves are there, so next-hop-self-all is not required.